backend/app/admin.py

```diff
@@ -13,10 +17,118 @@
     PhotographerAnalysisResult,
 )
 
+
+class MapSquareAdmin(admin.ModelAdmin):
+    """
+    MapSquare Admin
+    """
+    list_display = ('id', 'number', 'show_coordinates', 'count_photos', 'boundaries')
+    search_fields = ['id', 'number']
+    readonly_fields = ('all_photos',)
+
+    def show_coordinates(self, obj):
+        """
+        Returns coordinates of map square in (x,y) format
+        """
+        return '(' + obj.coordinates + ')'
+    show_coordinates.short_description = 'Coordinates'
+
+    @staticmethod
+    def count_photos(obj):
+        """
+        Returns an integer representing the number taken of that map square
+        """
+
+        # filter for all photo objects with obj's map square number
+        photo_obj = Photo.objects.filter(map_square__number=obj.number)
+        return len(photo_obj)
+    count_photos.short_description = 'Number of Photos'
+
+    @staticmethod
+    def all_photos(obj):
+        """
+        Displays all images taken at that map square
+        """
+        cmd = ''
+        # filter for all photo objects of obj map square
+        photo_obj = Photo.objects.filter(map_square__number=obj.number)
+        for photo in photo_obj:
+            # path to photo
+            obj_path = os.path.join(settings.LOCAL_PHOTOS_DIR, str(obj.number),
+                                    str(photo.number)+'_photo.jpg')
+
+            # path to photo's admin change page
+            photo_path = os.path.join('/admin/app/photo', str(photo.id))
+            cmd += '<a target = "blank" href = "{photo_url}"> <img src="{url}" ' \
```

Comment:
This is very (!!) minor but I wouldn't use the variable `cmd` here because this isn't a command — it's a link! Maybe html = '<a target="blank"....' etc

Reply:
Also no spaces around attribute setting in html is nicer :-) Like this:
<a target="blank" href="{photo_url}">